assignment2/q1.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <vector>
+using namespace std;
+
+struct Node{
+    int value;
+    Node *parent;
+    Node *left, *right;
+};
+
+Node *FindKey(Node *root, int key) {
+
+    if(root->value == key)
+        return root;
+
+    Node *l = NULL;
```

Comment:
No need to declare these early. You can declare them both inside the scope you use them:
```
if(root->left != NULL) {
  Node* l = FindKey(root->left, key);
  if(l != NULL)
    return l;
}
```

Then you can even give them the same name, e.g. `target` or something more indicative than `l`/`r`. Your choice :)